lib/goth/token.ex

```diff
@@ -303,7 +303,7 @@ defmodule Goth.Token do
   defp request(%{source: {:service_account, credentials, options}} = config)
        when is_map(credentials) and is_list(options) do
     url = Keyword.get(options, :url, @default_url)
-
+    sub = Keyword.get(options, :sub)
```

Comment:
I'm not 100% sure if calling this `sub` is ok semantically, we never set the JWT `sub` with this value. Perhaps something super explicit like:

```elixir
{:service_account, credentials, impersonate_service_account: email}
```

would be more appropriate?